api/app/Http/Controllers/ApiController.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: redjik
+ * Date: 16.07.15
+ * Time: 0:37
+ */
+
+namespace App\Http\Controllers;
+
+use App\Helpers\RouteHelper;
+use App\Repositories\BaseRepository;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\Request;
+use Spira\Repository\Model\BaseModel;
+use Spira\Responder\Responder\ApiResponder;
+use Symfony\Component\HttpFoundation\Response;
+
+abstract class ApiController
+{
+    /**
+     * Model Repository.
+     *
+     * @var BaseRepository
+     */
+    protected $repository;
+
+    /**
+     * @var ApiResponder
+     */
+    protected $responder;
+
+    /**
+     * Get all entities.
+     *
+     * @return Response
+     */
+    public function getAll()
+    {
+        return $this->responder->collection($this->repository->all());
+    }
+
+    /**
+     * Get one entity.
+     *
+     * @param  string $id
+     * @return Response
+     */
+    public function getOne($id)
+    {
+        return $this->responder->item($this->repository->find($id));
+    }
+
+    /**
+     * Post a new entity.
+     *
+     * @param  Request $request
+     * @return Response
+     */
+    public function postOne(Request $request)
+    {
+        $model = $this->repository->getModel();
+        $model->fill($request->all());
+        $this->repository->save($model);
+        $response = $this->responder->created();
+        if ($route = RouteHelper::getRoute($model)) {
+            $response->setContent(json_encode([$route]));
```

Comment:
Can you give an example? A `postOne` `User` ie `POST /users` should return something like

```
201 Created
{
    "_self" : "http://api.spira.io/users/856e4ca7-288c-4cdc-b825-d6328ca6f7dc"
}
```
The response is just a partial response of what the full `GET` request to that same `_self` url would return